packages/storage/amplify_storage_s3_dart/lib/src/model/s3_list_result.dart

```diff
@@ -58,38 +71,19 @@ class S3ListResult extends StorageListResult<S3Item> {
 
 /// The metadata returned from the Storage S3 plugin `list` API.
 class S3ListMetadata {
-  /// Creates a S3ListMetadata from the `commonPrefix` and `delimiter`
+  /// Creates a S3ListMetadata from the `delimiter`
   /// properties of the [s3.ListObjectsV2Output].
-  factory S3ListMetadata.fromS3CommonPrefixes({
-    List<s3.CommonPrefix>? commonPrefixes,
+  factory S3ListMetadata.fromDelimiter({
```

Comment:
instead of facoty method and a private ctor it may have a default ctor like below:
```
const S3ListMetadata({
    this.delimiter,
  });
  ```

Reply:
updated